### PR TITLE
[RFC] DDC-3976

### PR DESCRIPTION
**Pull request by guilhermeblanco**
Fixes #1542

### CODE REVIEW

File: tests/Doctrine/Tests/ORM/Functional/SecondLevelCacheManyToOneTest.php [hunk 1]
```diff
@@ -96,6 +96,26 @@ public function testPutAndLoadManyToOneRelation()
 
         $this->assertEquals($this->states[1]->getCountry()->getId(), $c4->getCountry()->getId());
         $this->assertEquals($this->states[1]->getCountry()->getName(), $c4->getCountry()->getName());
+
+        //evict collection on add
+        $prev  = $c3->getCities();
```
Comment (Ocramius) [inline tests/Doctrine/Tests/ORM/Functional/SecondLevelCacheManyToOneTest.php]: Please split into a separate test

**Comment by Ocramius** [comment]
Except for the test note, :+1:

File: lib/Doctrine/ORM/UnitOfWork.php [hunk 2]
```diff
@@ -826,10 +825,7 @@ private function computeAssociationChanges($assoc, $value)
         if ($value instanceof PersistentCollection && $value->isDirty()) {
             $coid = spl_object_hash($value);
 
-            if ($assoc['isOwningSide']) {
-                $this->collectionUpdates[$coid] = $value;
-            }
-
+            $this->collectionUpdates[$coid] = $value;
```
Comment (Ocramius) [inline lib/Doctrine/ORM/UnitOfWork.php]: Isn't this making also the inverse side of collections checked? :-\

File: lib/Doctrine/ORM/Cache/DefaultCollectionHydrator.php [hunk 3]
```diff
@@ -67,6 +67,7 @@ public function buildCacheEntry(ClassMetadata $metadata, CollectionCacheKey $key
         foreach ($collection as $index => $entity) {
             $data[$index] = new EntityCacheKey($metadata->name, $this->uow->getEntityIdentifier($entity));
         }
+
```
Comment (zeroedin-bill) [inline lib/Doctrine/ORM/Cache/DefaultCollectionHydrator.php]: Extra newline in unrelated file?

File: lib/Doctrine/ORM/UnitOfWork.php (hunk 2 above)
Comment (guilhermeblanco) [inline lib/Doctrine/ORM/UnitOfWork.php]: Yes, but you're indeed updating a collection adding/removing new items, so it should be checked depending of CollectionPersister implementation.
In most cases, it'll just return, except on CacheCollectionPersister, which would evict the collection (addressing the bug)

**Comment by nicolas-cajelli** [comment]
Hi @Ocramius @guilhermeblanco , i was just looking at the code, and i found out that this commit never reached the branch 2.5. Is there any reason for this?
We were discussing about this on #5821

**Comment by Ocramius** [comment]
Possibly didn't manage to backport, or @guilhermeblanco decided that the fix didn't need backporting or is too much effort to do so.

**Comment by nicolas-cajelli** [comment]
@Ocramius is there any guide to know how you guys are doing the backport? I would like to help to do it.

**Comment by Ocramius** [comment]
@nicolas-cajelli don't have a guide, but here's what I do:

```sh
git checkout 2.5
git checkout -b a-new-branch-that-backports-fix-#1234-to-2.5
git cherry-pick aaaaaaa1
git cherry-pick aaaaaaa2
git cherry-pick aaaaaaa3
git push origin a-new-branch-that-backports-fix-#1234-to-2.5
```
Then open a PR from that branch to 2.5
